demonstrations/tutorial_zne_catalyst.py

```diff
@@ -0,0 +1,282 @@
+leveraging just-in-time (JIT) compilation through
+`Catalyst <https://docs.pennylane.ai/projects/catalyst/en/stable/index.html>`_.
+
+What is ZNE
+-----------
+Zero-Noise Extrapolation (ZNE) is a technique used to mitigate the effect of noise on quantum
+computations. First introduced in [#temme2017zne]_, it helps improve the accuracy of quantum
+results by running circuits at varying noise levels and extrapolating back to a hypothetical
+zero-noise case. While this tutorial won't delve into the theory behind ZNE in detail, let's first
+review what happens when using the protocol in practice.
+
+Stage 1: Generating Noise-Scaled Circuits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In its digital version [#DZNEpaper]_, ZNE works by generating circuits with **increased** noise. 
+Currently, ZNE in Catalyst supports two methods for generating noise-scaled circuits:
+
+1. **Global folding**: If a circuit implements a global unitary :math:`U`, global folding applies
+   :math:`U(U^\dagger U)^n` for some integer :math:`n`,
+   effectively scaling the noise in the entire circuit.
+2. **Local folding**: Individual gates are repeated (or folded) in contrast with the entire
+   circuit.
+
+Stage 2: Running the circuits
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once noise-scaled circuits are created, they need to be run! These can be executed on either real
+quantum hardware or a noisy quantum simulator. In this tutorial, we'll use the
+`Qrack quantum simulator <https://qrack.readthedocs.io/>`_, which is both compatible with Catalyst,
+and implements a noise model. For more about the integration of Qrack and Catalyst, see
+the demo `QJIT compilation with Qrack and Catalyst <https://pennylane.ai/qml/demos/qrack/>`_.
+
+Stage 3: Combining the results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+After executing the noise-scaled circuits, an extrapolation on the results is performed  
+to estimate the zero-noise limit---the result we would expect in a noise-free scenario. 
+Catalyst provides two methods for perfoming this extrapolation:
+
+1. **Polynomial extrapolation**, and
+2. **Exponential extrapolation**.
+
+Using ZNE with Pennylane
+------------------------
+The demo `Error mitigation with Mitiq and PennyLane <https://pennylane.ai/qml/demos/tutorial_error_mitigation/>`_
+shows how ZNE, along with other error mitigation techniques, can be carried out in Pennylane by using Mitiq, 
+a Python library developed by Unitary Fund.
+
+ZNE in particular is also offered out of the box in Pennylane as a *differentiable* error mitigation technique,
+for usage in combination with variational workflows. More on this in the tutorial 
+`Differentiating quantum error mitigation transforms <https://pennylane.ai/qml/demos/tutorial_diffable-mitigation/>`_.
+
+
+Using ZNE in Catalyst
+---------------------
+On top of the error mitigation routines offered in Pennylane, ZNE is also available for just-in-time 
+(JIT) compilation, starting from Catalyst v0.8.1.
```

Comment:
```suggestion
On top of the error mitigation routines offered in Pennylane, ZNE is also available for just-in-time 
(JIT) compilation.
```
(since the tutorial is 'time invariant')

Reply:
I agree that the tutorial is 'time invariant', but it won't work if the users have Catalyst v<0.8.1

Reply:
Note that since we'll have to make another release, we should update this to 8.2 (if we plan on keeping it).